Add a gem configurator to enable changing target inky template engine

bobmaerten opened the pull request:
Paving the way for other configuration variables, this PR implements
the "Configurator Pattern".

To change default ERB template engine, set an initializer like this:
```ruby
# config/initializers/inky.rb
Inky.configure do |config|
  config.template_engine = :slim
end
```

Of course, you have to ensure this template is loaded in you app.

Rails generator is amended to output Slim or Haml versions of mailer_layout.

Also updates README files concordantly, and remove remaining references to
REXML after Nokogiri replacement.

Fixes #14

kball commented on lib/generators/inky/install_generator.rb:
```diff
@@ -6,11 +6,12 @@ class InstallGenerator < ::Rails::Generators::Base
         return nil unless layout_name == 'mailer'
 
-        original_mailer = File.join(layouts_base_dir, 'mailer.html.erb')
+        original_mailer = File.join(layouts_base_dir, "mailer.html.#{extenstion}")
```
Looks like there's a typo in the extension

bobmaerten replied:
Thanks for spotting this! sending a fix.

kball commented:
Awesome!  Thanks for the PR!  I'll try to get this tested out and merged shortly.

bobmaerten commented:
I have a problem with this configuration when I have a slim partial using many `render`, only the first is actually rendered. I'll try to revert to ERB to see if it's slim related.

bobmaerten commented:
I have similar behavior with erb partials, dammit! I have no clue for this russian doll rendering problem...

kball commented:
@bobmaerten can you explain a little more about the case that is causing problems?  Also, check if it is causing issues for the master version as well - may be independent of this PR.

bobmaerten commented:
Sure.

Consider this inky view:
```erb
<row>
  <columns>
    <%= render "intro" %>
    Hello, world!
  </columns>
</row>
```

and these partials
`_intro.html.inky`
```erb
<p>Yo!</p>
<%= render "sub_intro" %>
```

`_sub_intro.html.inky`
```erb
<%= render "first" %>
<%= render "second" %>
```

When I render the view, the `second` partial is not rendered.

I'll try to test it out of master and develop branches (I have to convert many views and layout to erb for that :/)

bobmaerten commented:
I tried with `master` branch and erb/inky templates, but apart from one #11 error (multiple root) which I fixed with a `<wrapper>` tag, all my partials are rendering correctly.

So I tested with `develop` branch and it worked!

So I converted my views/partials/layout back to slim with this branch and it worked too!

In the end it was "just" a Inky parsing error (which I do not quite understand). I suspect Nokogiri from `develop` branch to discard parsing when there is a error, without notification.

bobmaerten commented:
From the example above, in order to full partials rendering, I had to "wrap" in `_sub_intro.html.inky` file:
```erb
<wrapper>
  <%= render "first" %>
  <%= render "second" %>
</wrapper>
```

marcandre commented:
I think the multiple render issue is fixed in the latest release.

A config for the default rendered (default: `:erb`) is a fine idea. Doesn't solve the issue for anyone with mixed sources though, but it's a good start.

marcandre commented:
Great PR. @bobmaerten I've rebased and improved your PR, would be nice if you could have a look at #37

bobmaerten commented:
Thanks for rebasing this! I was not so eager to do this as our forked repo runs quite correctly but if you're heading this way, this might be a good way to dig into this gem again.

marcandre commented:
Merged [#37]
